### DISCUSSION TITLE
GCP 인스턴스 미사용 상황에서 발생한 트래픽 이슈

### DISCUSSION THREAD

**Discussion by seokjunKing** [WHATDO-TEAM/whatdo#40] [트러블슈팅]
어제 저녁 깜박하고 vm인스턴스를 중지를 안했더니,

![1](https://github.com/user-attachments/assets/ecf468e8-0a0b-484a-8f01-f3c21dda7d3a)
![2](https://github.com/user-attachments/assets/87efe6b6-a9ac-431e-a62a-8a5731fe5241)

서비스 약관 위반이라고 사용할 수가 없게 되었습니다..

----

확인을 해보니 암호화폐 채굴로 예상되는 흔적이 발견되었대요
참고로 저 시간대에는 인스턴스 연결도 안되어 있었습니다.

그래서 찾아보니 재검토 요청을 하면 다시 정상적으로 사용할 수 있게 해준다고 하더라구요.

![4](https://github.com/user-attachments/assets/c95797af-5ad0-4e2e-88f4-4f0bd315b2e1)

------

하지만 이메일 답변으로 구체적인 상황에 대해 설명을 다시 요청을 했습니다.

그래서 로그를 확인해보고 gpt에 분석을 요청했습니다.

![3](https://github.com/user-attachments/assets/ecec41ef-d60e-4458-a2e9-296d8669426a)

**분석 내용**

로그 내용을 보면 compute.googleapis.com/shielded_vm_integrity와 관련된 메시지가 기록된 것 같습니다. 이 메시지는 GCP의 Shielded VM 기능과 연관된 것으로 보이는데, 이 기능은 VM의 보안 상태를 모니터링하고 무결성을 보장하기 위해 사용됩니다. Shielded VM의 무결성 검사 트래픽이 발생했을 가능성이 있습니다.

Shielded VM의 무결성 확인 프로세스는 다음과 같은 이유로 트래픽을 유발할 수 있습니다:

자동 보안 검사 및 로그 기록: Shielded VM은 주기적으로 VM의 무결성을 점검하여 보안이 유지되고 있는지 확인합니다. 이 과정에서 트래픽이 발생하고 해당 내역이 compute.googleapis.com/shielded_vm_integrity 로그에 기록됩니다.

리전별 보안 점검: asia-northeast2와 같은 특정 리전에서 무결성 검사가 발생했다는 정보가 포함되어 있는 점으로 보아, VM이 배치된 리전 내에서 주기적인 보안 확인이 있었을 수 있습니다.

이 로그가 의미하는 것은 주기적인 보안 검사가 이루어졌다는 것으로, 실제로 사용자가 VM을 활성화한 상태가 아니더라도 시스템에서 기본적으로 수행되는 검사가 트래픽으로 기록된 것입니다. Shielded VM의 무결성 검사는 자동으로 수행되며, 별도의 설정을 하지 않아도 활성화될 수 있습니다.

-------

현재는 gpt의 정보를 바탕으로 이의신청을 해놓은 상황입니다.

**Comment by seokjunKing** [comment]
11-06 13시경 다른 인스턴스에서 airflow 테스트 진행 중 인스턴스가 다시 정지되는 상황이 일어났습니다.
이유는 같은 이유였는데
지난번 복구신청 후 답변이 안와서 다시 복구 신청을 놓았습니다.

---
![tqqr](https://github.com/user-attachments/assets/5579e4fd-ca0b-4a4c-9cd9-123adad62a03)


오늘 오전에 확인해보니 두 인스턴스 모두 복구가 되었습니다.

**Comment by Kdreamtomaster** [comment]
저도 저러한 메일을 받고 
이전에 사용했던 제가 작업중이던 spark-master instance에도 CPU 100% 유지를 2일 정도 했던 의심 정황이 있었고
일단 저희 프로젝트 내의 전 인스턴스 백업후 삭제를 하였습니다. 

1. 기존 키파일 전부 삭제 (메타데이터 전부 삭제) -> 누출되었을 가능성 있어 삭제하였습니다.
2. 코드, 설정 백업이 완료된 인스턴스의 경우 삭제 (전부 삭제후 재구성 예정입니다.) -> 특정 채굴 코드가 있을 가능성이 있어 삭제하였습니다. 
3. 문제가 될 수 있는 보안 규칙 전부 삭제 및 보안 규칙 관련 팀내 룰 전파 예정입니다. -> 느슨한 방화벽 규칙(0.0.0.0/0, 모든 트래픽 규칙 삭제)
4. 구글 계정 2단계 인증 재확인 중 -> 모두 확인 되었습니다. 별다른 이상징후 없었습니다.
